Kernel/generate-version-file.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+OUTPUT_FILE=$1
+GIT_HASH=$( (git log --pretty=format:'%h' -n 1 | cut -c1-7) || true )
```

Comment:
Can we have a sensible default when `git` is not available, or the source code is not build with a `.git` directory present? (E.g. an export of the code)

Reply:
Also: you might want to take into account whether we're building from unmodified source code, or a repository with outstanding changes (either in the working tree or index).